libnavui-dropin/src/main/res/layout-land/mapbox_navigation_view_layout.xml

```diff
@@ -58,14 +58,30 @@
                     tools:layout_height="120dp"
                     tools:background="@color/mapbox_main_maneuver_background_color" />
 
+                <FrameLayout
+                    android:id="@+id/compassLayout"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    app:layout_constraintTop_toTopOf="parent"
+                    app:layout_constraintStart_toEndOf="@id/guidanceLayout"
+                    />
+
+                <FrameLayout
+                    android:id="@+id/scalebarLayout"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    app:layout_constraintTop_toTopOf="parent"
+                    app:layout_constraintStart_toEndOf="@id/guidanceLayout"
+                    />
+
                 <FrameLayout
                     android:id="@+id/speedLimitLayout"
                     android:layout_width="75dp"
                     android:layout_height="86dp"
                     android:layout_marginStart="4dp"
                     android:layout_marginTop="8dp"
                     app:layout_constraintStart_toEndOf="@id/guidanceLayout"
-                    app:layout_constraintTop_toTopOf="@id/guidanceLayout"
+                    app:layout_constraintTop_toBottomOf="@id/scalebarLayout"
```

Comment:
> constraintTop_toBottomOf="@id/scalebarLayout" with marginTop=8dp - when scalebar is visible.
constraintTop_toTopOf="parent" with marginTop=8dp - when scalebar is gone

But it's the same. If the scalebar is gone we can leave `constraintTop_toBottomOf="@id/scalebarLayout"` and it will mean `constraintTop_toTopOf="parent"`. Because scalebarLayout is constrained to parent's top.

Reply:
Oh yeah! You are correct.

Reply:
Actually, I think it's accounted for. See https://github.com/mapbox/mapbox-navigation-android/pull/6355/files#diff-29aeae6515054485659a1bcb7b2f41ed16a2dc6cdd35bf443a4f22daca71a67fR30

`constraintTop_toTopOf="@id/guidanceLayout"` means the same as `constraintTop_toBottomOf="@id/scalebarLayout"` if `scalebarLayout` is gone. And its visibility is changed depending on whether we have ManeuverView or not.

The only thing left is the marginTop. You say it should be zero when maneuver view is visible. But it's hardcoded in the xml and we don't change it regardless of this PR. I think it's OK. Attaching the screenshots.
<img width="827" alt="image" src="https://user-images.githubusercontent.com/32109537/191690988-246337a6-9e7c-434a-b32f-b602ce1cd43f.png">
<img width="828" alt="image" src="https://user-images.githubusercontent.com/32109537/191691063-2c4225ec-38d0-458a-98f7-ea9bc4e2fe69.png">

Reply:
Take a look at the action buttons. They have no margin at all when compass is disabled
![Screenshot_2022-09-22-13-47-29-05_ab98132dc56698913f44974ecb476e75](https://user-images.githubusercontent.com/9770186/191696116-fd3cefa1-7275-4022-8c92-ee14f1ed67b1.jpg)